diff --git a/doomgeneric/i_video.c b/doomgeneric/i_video.c
index 84bebde..89ec478 100644
--- a/doomgeneric/i_video.c
+++ b/doomgeneric/i_video.c
@@ -239,9 +239,9 @@ void I_InitGraphics (void)
 		s_Fb.red.length = 8;
 		s_Fb.transp.length = 8;
 
-		s_Fb.blue.offset = 0;
+		s_Fb.red.offset = 0;
 		s_Fb.green.offset = 8;
-		s_Fb.red.offset = 16;
+		s_Fb.blue.offset = 16;
 		s_Fb.transp.offset = 24;
 	}
 
